### PR TITLE
#17 Match ui implemented

### PR DESCRIPTION
**Pull request by gabriellesc**

### CODE REVIEW

**Comment by michellemtchai** [comment]
1. Stop Docker if it's running
2. Then, I ran the following commands:
  - `docker-compose down`
  - `docker-compose up`
  - `docker-compose run -eRAILS_ENV=development rails-app rake db:drop db:create db:migrate db:seed`
  - `docker-compose run -eRAILS_ENV=development rails-app rake db:seed:chass`

I had to run ChassImporter because the first time the site ran it didn't have any data in it.

3. Some issues I noticed:
  - On the top menu bar, `All Assigned` and `All Unassigned` doesn't get highlighted when clicked. Though, the routes look to be right.
screenshot from 2017-06-23 10-30-46
screenshot from 2017-06-23 10-31-40

  - The highlight in the sidebar for the `Applicants by Course` view works, but I had to try several refresh to get it to load the panels. Basically it sometimes loads and sometimes does not. There doesn't seem to be a specific pattern of when it would work. Though, the click seem to work more in Firefox than Chrome. (Maybe it's a compatibility issue?)

- The close button for the right-most panel tend to disappear in a 3-panel view.
screenshot from 2017-06-23 10-44-00

- `Applicants by Course` can switch fine to `Courses`, but when I'm on `Courses` and I click on any of the other link, the `Course` view stays. For example, if I want to get the `Applicants by Course` view to load, I would have to refresh the page after clicking on the `Applicant by Course` button on the menu. Any switching from `Applicants by Courses` to any route but `Courses` seem to require a reload as well. Basically, switching between the "tabs" requires a lot of reloading, but sometimes it doesn't. I'll try to figure out the pattern.

- Both `All Assigned` and `All Unassigned` loaded as empty pages. There's no template like in `Courses` and `Summary`.

**Comment by gabriellesc** [comment]
> * On the top menu bar, All Assigned and All Unassigned doesn't get highlighted when clicked. Though, the routes look to be right.
> * Both All Assigned and All Unassigned loaded as empty pages. There's no template like in Courses and Summary.

Yes, those two views don't render because I wrote the top level components for those views (`Assigned` in app/components/assigned.js and `Unassigned` in app/components/unassigned.js) as templates. That is, in the `Assigned` and `Unassigned` render methods, I put in the components that would make up those views, but I didn't pass the necessary props to those components. Because the view component isn't being rendered, the tab highlighting also doesn't happen.  
Filling those in will be tasks for the next sprint, I think.

> * The highlight in the sidebar for the Applicants by Course view works, but I had to try several refresh to get it to load the panels. Basically it sometimes loads and sometimes does not. There doesn't seem to be a specific pattern of when it would work. Though, the click seem to work more in Firefox than Chrome. (Maybe it's a compatibility issue?)
> * Applicants by Course can switch fine to Courses, but when I'm on Courses and I click on any of the other link, the Course view stays. For example, if I want to get the Applicants by Course view to load, I would have to refresh the page after clicking on the Applicant by Course button on the menu. Any switching from Applicants by Courses to any route but Courses seem to require a reload as well. Basically, switching between the "tabs" requires a lot of reloading, but sometimes it doesn't. I'll try to figure out the pattern.

I haven't been able to replicate either of these issues.

> * The close button for the right-most panel tend to disappear in a 3-panel view.

This is an issue I'm aware of and is on my list to fix, but is not a priority for this application. Karen has said that she is not likely to display more than 2 courses simultaneously.

File: app/controllers/app_controller.rb~ [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class AppController < ActionController::Base
```
Comment (freeatnet) [inline app/controllers/app_controller.rb~]: Temporary file

File: app/assets/stylesheets/abc.css [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    border-right: 5px solid rgba(255, 255, 255, 0);
+    cursor: col-resize;
+}
+
+.Resizer.vertical:hover {
+    border-left: 5px solid rgba(0, 0, 0, 0.5);
+    border-right: 5px solid rgba(0, 0, 0, 0.5);
+}
+
+.well {
```
Comment (freeatnet) [inline app/assets/stylesheets/abc.css:50]: Let's avoid changing bootstrap classes. Maybe we can make this `.well-centered.well` or something?

File: app/assets/stylesheets/abc.css [hunk 3]
```diff
@@ -0,0 +1,53 @@
+nav+.SplitPane {
```
Comment (freeatnet) [inline app/assets/stylesheets/abc.css:1]: This had me puzzled for a moment. Good choice of selector though!

**Review by freeatnet — APPROVED** [review]
Minor piece of cleanup suggested. If you're happy with this branch, :shipit:

**Pull request merged**